src/adagrams.js

```diff
@@ -1,8 +1,164 @@
 const Adagrams = {
+  allLetters: {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  },
   drawLetters() {
-    // Implement this method for wave 1
+    let letterPool = [];
+
+    Object.keys(this.allLetters).forEach(letter => {
+
+      let times = this.allLetters[letter];
+      for (let i = 0; i < times; i += 1) {
+        letterPool.push(letter);
+      }
+    });
+    letterPool = this.shuffle(letterPool);
+    const hand = letterPool.slice(0, 10);
+    // console.log(hand);
+    return hand;
+  },
+  shuffle(array) {
+    let currentIndex = array.length;
+    let temporaryValue;
+    let randomIndex;
+
+    while (0 !== currentIndex) {
+
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+
+      temporaryValue = array[currentIndex];
+      array[currentIndex] = array[randomIndex];
+      array[randomIndex] = temporaryValue;
+    }
+
+    return array;
+  },
+  usesAvailableLetters(word, drawn) {
+    let drawnCopy = [...drawn];
+    for (let i = 0; i < word.length; i += 1) {
+      let char = word.charAt(i);
+
+      if (drawnCopy.includes(char)) {
+        let index = drawnCopy.indexOf(char);
+        drawnCopy.splice(index, 1);
+      } else {
+        return false;
+      }
+    }
+    return true;
   },
+  scoreWord(word) {
+    let score = 0;
+
+    if (word === '') {
+      return score;
+    }
+
+    if (word.length >= 7) {
+      score += 8;
+    }
+
+    for (let i = 0; i < word.length; i += 1) {
+      switch (word.charAt(i).toUpperCase()) {
+        case 'A':
+        case 'E':
```

Comment:
I'm not a big fan of organizing the letter scores in a `switch` statement like this. It's not very DRY (you've had to type `case` 26 times), but more importantly the data is intertwined with the code that uses it, which means there's no easy way to reuse this information.

For example, what if the player had a hand full of tiles, and you wanted to be able to display the value of each of those letters? You would have to copy/paste this switch in that code, or at least extract it to a method.

A better approach would be to store the scores in a hash:

```javascript
const LETTER_VALUES = {
  a: 1,
  b: 3,
  // ...
}
```

then here you could use it like this:

```javascript
word.forEach((letter) => {
  const value = LETTER_VALUES[letter.toLowerCase()];
  if (value) {
    totalScore += letter.toLowerCase();
  } else {
    throw new Error(`Invalid character ${letter}`);
  }
}
```